.agents/architecture/ADR-032-exit-code-standardization.md

```diff
@@ -0,0 +1,309 @@
+
+**Pros**:
+
+- Simple
+- Already common in some scripts
+
+**Cons**:
+
+- No diagnostic value (caller cannot distinguish failure types)
+- Debugging requires log inspection
+- Cannot implement conditional retry logic
+
+### Option 3: HTTP-Style Codes
+
+| Code | Meaning |
+|------|---------|
+| 0 | Success (200-299 equivalent) |
+| 4 | Client error (400-499 equivalent) |
+| 5 | Server error (500-599 equivalent) |
+
+**Pros**:
+
+- Familiar to web developers
+
+**Cons**:
+
+- Awkward mapping (exit codes are not HTTP codes)
+- Loses granularity (many error types map to 4 or 5)
+- Not conventional for CLI tools
+
+### Option 4: No Standard (Status Quo)
+
+**Pros**:
+
+- No migration effort
+
+**Cons**:
+
+- Continued inconsistency
+- Cross-language bugs persist
+- Test reliability issues continue
+- Debugging remains difficult
+
+---
+
+## Decision Outcome
+
+**Chosen option: Option 1 - POSIX-Style Standard**
+
+### Exit Code Reference
+
+| Code | Category | Semantic Meaning | When to Use |
+|------|----------|------------------|-------------|
+| 0 | Success | Operation completed successfully | All success paths, including idempotent no-ops |
+| 1 | Logic Error | General failure, validation failed | Business logic failures, assertion violations |
+| 2 | Config Error | Usage, configuration, or environment error | Missing params, invalid args, missing dependencies |
+| 3 | External Error | External service or API error | GitHub API failures, network errors, timeouts |
+| 4 | Auth Error | Authentication or authorization failure | Token expired, permission denied, rate limited |
+| 5-99 | Reserved | Future standard use | Do not use until standardized |
+| 100+ | Script-Specific | Custom error codes | Document in script header comment |
+
+### Documentation Requirement
+
+All scripts MUST include exit code documentation in the script header:
+
+```powershell
+<#
+.SYNOPSIS
+    Brief description
+
+.NOTES
+    EXIT CODES:
+    0  - Success: Operation completed
+    1  - Error: Validation failed (specify conditions)
+    2  - Error: Missing required parameter
+    3  - Error: GitHub API returned error
+    4  - Error: Authentication failed
+
+    When called from bash/CI:
+    - Exit 0 = success ($? equals 0)
+    - Exit non-zero = failure ($? non-zero)
+#>
+```
+
+### Rationale
+
+1. **Industry Alignment**: POSIX exit codes and sysexits.h provide proven conventions
+2. **Diagnostic Value**: Operators can triage without logs (2=fix config, 3=check GitHub, 4=renew token)
+3. **Retry Logic**: CI can implement intelligent retry (retry on 3, fail fast on 2)
+4. **Test Clarity**: Pester tests assert specific codes, improving test documentation
+
+### Enforcement
+
+1. **This ADR**: Canonical reference
+2. **Code Review**: Verify exit code documentation and compliance
+3. **Pester Tests**: Include exit code assertions with semantic checks
+4. **PSScriptAnalyzer**: Future custom rule for undocumented exit statements
+
+---
+
+## Migration Plan

```

Comment:
@rjmurillo-bot Link the GitHub issues to perform this work. Create one new epic with three sub-tasks, one for each phase